packages/extension-base/src/services/history-service/index.ts

```diff
@@ -74,6 +90,52 @@ export class HistoryService implements StoppableServiceInterface, PersistDataSer
     return this.historySubject;
   }
 
+  private async fetchSubscanTransactionHistory (chain: string, address: string) {
+    const result: TransactionHistoryItem[] = [];
+
+    // todo: optimise this: reduce unnessary query
+    const extrinsicItems = await this.subscanService.getAllExtrinsicItems(chain, address);
+    const transferItems = await this.subscanService.getAllTransferItems(chain, address);
+    const chainInfo = this.chainService.getChainInfoByKey(chain);
+
+    const excludeTransferExtrinsicHash: string[] = [];
+
+    extrinsicItems.forEach((x) => {
+      excludeTransferExtrinsicHash.push(x.extrinsic_hash);
+
+      if (supportedExtrinsicParser.includes(getExtrinsicParserKey(x))) {
+        result.push(parseSubscanExtrinsicData(address, x, chainInfo));
+      }
+    });
+
+    transferItems.forEach((t) => {
+      if (!excludeTransferExtrinsicHash.includes(t.hash)) {
+        result.push(parseSubscanTransferData(address, t, chainInfo));
+      }
+    });
+
+    await this.addHistoryItems(result);
+
+    return result;
+  }
+
+  async subscribeHistories (chain: string, address: string, cb: (items: TransactionHistoryItem[]) => void) {
```

Comment:
Please return data or callback onetime in this api instead of return only unsub call